custom_components/windcentrale/sensor.py

```diff
@@ -157,6 +158,9 @@ def update(self):
                 self._state = self._windturbine.live_data[self._sensor] / self._windturbine.total_shares * self._windturbine.shares
             elif self.type == "runpercentage":
                 self._state = round(timedelta(hours=self._windturbine.live_data[self._sensor]) / (datetime.now() - datetime(datetime.now().year, 1, 1)) * 100, 2)
+            elif self.type == "yearproducedpercentage":
+                """ Year production / total_shares * 2 (one share ~ 0.5MW) / 1000 (convert from kWh normal) * 100% """
+                self._state = round(self._windturbine.live_data[self._sensor] / self._windturbine.total_shares * 0.2, 1)
```

Comment:
I don't think this is the best way to calculate the prognosis.
I would take the values ​​from the `energy`  sensor and divided that by the total kWh the windturbine is raided for.

So example:
The witte juffer is now at 2.053.800 kWh
The witte juffer is raided to do 2.800.000 kWh every year.

So In code it will look like this:
```python 
self._state = round((self._windturbine.live_data[self._sensor] / 2800000) * 100, 2)
```

With your calculation the answer is `71,8 %`
With my calculation the answer is `73,35 %`

Reply:
In the previous app, the prognoses was about the windshares which were targetted at 500kWh /year.
The prognoses is about that part... 

I reality, the windmill might generate a bit more (or less) energy, but that isn't flowing to the shareholders, as far as i know. Hence the calculation through the year-production divided by the amount of shares of a windmill.